lld/MachO/Driver.cpp

```diff
@@ -1295,6 +1394,27 @@ static void createFiles(const InputArgList &args) {
       break;
     }
   }
+
+  if (config->readThreads) {
+    multiThreadedPageIn(deferredFiles);
+
+    DeferredFiles archiveContents;
+    std::vector<ArchiveFile *> archives;
+    for (auto &file : deferredFiles)
+      if (ArchiveFile *archive = dyn_cast<ArchiveFile>(
+              processFile(file.buffer, &archiveContents, file.path,
+                          LoadType::CommandLine, file.isLazy)))
+        archives.push_back(archive);
```

Comment:
In my opinion, having the `processFile` in a separate line makes it clear that it will always happen, even if they are not `ArchiveFile`.

```suggestion
      auto inputFile = processFile(file.buffer, &archiveContents, file.path,
                          LoadType::CommandLine, file.isLazy);
      if (ArchiveFile *archive = dyn_cast<ArchiveFile>(inputFile))
        archives.push_back(archive);
```